warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IsNotNullPruningVisitor.java

```diff
@@ -188,6 +270,26 @@ protected String fieldForChild(JexlNode node) {
         return null;
     }
     
+    /**
+     * Determine if this node is a <code>filter:includeRegex</code> function, and extract fields
+     *
+     * @param node
+     *            an ASTFunction node
+     * @return a single field if it exists, otherwise null
+     */
+    protected String fieldForFunction(JexlNode node) {
+        FunctionJexlNodeVisitor visitor = new FunctionJexlNodeVisitor();
+        node.jjtAccept(visitor, null);
+        
+        if (visitor.namespace().equals("filter") && visitor.name().equals("includeRegex")) {
```

Comment:
This seems like a really specific case.  Can't we do something generic for functions in general?  I think the function descriptor classes have a method to return the fields.

Reply:
To limit scope I would like to only add the ability to prune if an `isNotNull` term is intersected with a `filter:includeRegex` term against the same field (for example, `filter:isNotNull(FOO) && filter:includeRegex(FOO, 'ba.*')`.

I'm open to adding more cases in the future, did you have any specific functions in mind? An initial list is below
- content function: primarily index-only
- f:includeText once that move is complete
- f:geo functions?

Reply:
I think you can do it for all functions.  The descriptor mechanism gives you a way to do that without having to enumerate them.

Reply:
The descriptor `fields()` method requires the MetadataHelper and the datatype filter for the query. I was hoping to avoid adding those classes to this visitor to keep it lightweight. 

Even so, `filter:isNotNull` is not allowed to have to index-only fields as arguments. Logically, we can't prune against content functions because content functions are (almost?) always run against index-only fields.

Reply:
Wrote up an issue for this https://github.com/NationalSecurityAgency/datawave/issues/1605